discovery-provider/src/tasks/index.py

```diff
@@ -125,36 +126,58 @@ def initialize_blocks_table_if_necessary(db: SessionManager):
     return target_blockhash
 
 
-def get_latest_block(db: SessionManager):
-    latest_block = None
+def get_latest_block(db: SessionManager, retry: bool = False):
     block_processing_window = int(
         update_task.shared_config["discprov"]["block_processing_window"]
     )
-    with db.scoped_session() as session:
-        current_block_query = session.query(Block).filter_by(is_current=True)
-        assert current_block_query.count() == 1, "Expected SINGLE row marked as current"
+    try:
+        with db.scoped_session() as session:
+            current_block_query = session.query(Block).filter_by(is_current=True)
+            assert (
+                current_block_query.count() == 1
+            ), "Expected SINGLE row marked as current"
 
-        current_block_query_results = current_block_query.all()
-        current_block = current_block_query_results[0]
-        current_block_number = current_block.number
+            current_block_query_results = current_block_query.all()
+            current_block = current_block_query_results[0]
+            current_block_number = current_block.number
 
-        if current_block_number == None:
-            current_block_number = 0
+            if current_block_number == None:
+                current_block_number = 0
 
-        target_latest_block_number = current_block_number + block_processing_window
+            target_latest_block_number = current_block_number + block_processing_window
 
-        latest_block_from_chain = update_task.web3.eth.getBlock("latest", True)
-        latest_block_number_from_chain = latest_block_from_chain.number
+            latest_block_from_chain = update_task.web3.eth.getBlock("latest", True)
+            latest_block_number_from_chain = latest_block_from_chain.number
 
-        target_latest_block_number = min(
-            target_latest_block_number, latest_block_number_from_chain
-        )
+            target_latest_block_number = min(
+                target_latest_block_number, latest_block_number_from_chain
+            )
 
-        logger.info(
-            f"index.py | get_latest_block | current={current_block_number} target={target_latest_block_number}"
-        )
-        latest_block = update_task.web3.eth.getBlock(target_latest_block_number, True)
-    return latest_block
+            logger.info(
+                f"index.py | get_latest_block | current={current_block_number} target={target_latest_block_number}"
+            )
+            latest_block = update_task.web3.eth.getBlock(
+                target_latest_block_number, True
+            )
+
+            # if the block had no transactions, retry after small delay to confirm the block is actually empty
```

Comment:
net new code for retrying the function if no transactions

Reply:
i thought about this a bit. this actually should never happen because there should be at least 1 tx for each block (the AuRa one, my guess is some validators are not running the correct version or something). but given our volume and tx's and the occurence of 0 tx blocks i don't think this should introduce significant latency to the process but i'm gonna hotfix this on prod to observe before releasing

Reply:
i was just looking here https://blockscout.com/poa/core/blocks and see like 8 in the last 4 min so its def non-zero

Reply:
yeah it's roughly 15% but weighed against the possibility of divergence and missing content this seemed at least feasible to try, especially considering the latency of this call is so low compared to block production time